### PR TITLE
[3.3.5] Core/Scripts/DB: move core script 'npc_parqual_fintallas' to SAI

### PR DESCRIPTION
**Pull request by ghost**
Additional change: menu options are changed from containing only names
to include the text " is my answer.", based on the lines in `broadcast_text`.
- replaced core text with DB SAI script
- created DB creature_text entries for Parqual Fintallas
- created DB gossip_menu_option entries for the NPC
- implemented NPC say lines in reply to all the answers
- added conditions for which gossip should show, depending on quest status
- added conditions to avoid exploits in the time it takes for NPC to say his texts

This PR is tested several times on a horde character (horde-only quest).
It now works as originally intended, with option-specific NPC replies.

### CODE REVIEW

File: sql/updates/world/2016_01_25_09_world.sql [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+(4488, 0, 13, 14, 61, 0, 100, 0,    0,    0, 0, 0,  1,    3, 5000, 0,0,0,0, 7, 0,0,0,0,0,0,0,"Parqual Fintallas - on Gossip Option 3 selected - Say Text 3"),
+(4488, 0, 14, 15, 61, 0, 100, 0,    0,    0, 0, 0, 15, 6628,    0, 0,0,0,0, 7, 0,0,0,0,0,0,0,"Parqual Fintallas - on Gossip Option 3 selected - Credit Quest 6628 'Test of Lore'"),
+(4488, 0, 15,  0, 61, 0, 100, 0,    0,    0, 0, 0,  5,    2,    0, 0,0,0,0, 7, 0,0,0,0,0,0,0,"Parqual Fintallas - on Gossip Option 3 selected - Play Emote 2 (ONESHOT_BOW)");
+
+-- conditions showing gossip menu and gossip menu options only when Quest 6628 is active
+DELETE FROM `conditions` WHERE `SourceTypeOrReferenceId`= 14 AND `SourceGroup`= 4764 AND `SourceEntry`= 5822;
+DELETE FROM `conditions` WHERE `SourceTypeOrReferenceId`= 15 AND `SourceGroup`= 4764 AND `SourceEntry` IN (0,1,2,3,4);
+INSERT INTO `conditions` (`SourceTypeOrReferenceId`,`SourceGroup`,`SourceEntry`,`SourceId`,`ElseGroup`,`ConditionTypeOrReference`,`ConditionTarget`,`ConditionValue1`,`ConditionValue2`,`ConditionValue3`,`ErrorTextId`,`ScriptName`,`Comment`) VALUES
+(14, 4764, 5822, 0, 0, 9, 0, 6628, 0, 0, 0, '', 'Show gossip dialog text 5822 if Quest 6628 is Active (taken)'),
```
Comment (joschiwald) [inline sql/updates/world/2016_01_25_09_world.sql]: you need a condition for both texts

File: sql/updates/world/2016_01_25_09_world.sql [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+(4488, 0,  5,  0, 61, 0, 100, 0,    0,    0, 0, 0,  1,    1, 5000, 0,0,0,0, 7, 0,0,0,0,0,0,0,"Parqual Fintallas - on Gossip Option 1 selected - Say Text 1"),
+(4488, 0,  6,  0, 52, 0, 100, 0,    1, 4488, 0, 0,  1,    4, 3000, 0,0,0,0, 7, 0,0,0,0,0,0,0,"Parqual Fintallas - on Text 1 Over - Say Text 4"),
+(4488, 0,  7,  0, 52, 0, 100, 0,    4, 4488, 0, 0, 11, 6767,    0, 0,0,0,0, 7, 0,0,0,0,0,0,0,"Parqual Fintallas - on Text 4 Over - Cast spell 6767 'Mark of Shame'"),
+(4488, 0,  8,  9, 62, 0, 100, 0, 4764,    2, 0, 0, 72,    0,    0, 0,0,0,0, 7, 0,0,0,0,0,0,0,"Parqual Fintallas - on Gossip Option 2 selected - Close Gossip"),
+(4488, 0,  9,  0, 61, 0, 100, 0,    0,    0, 0, 0,  1,    2, 5000, 0,0,0,0, 7, 0,0,0,0,0,0,0,"Parqual Fintallas - on Gossip Option 2 selected - Say Text 2"),
+(4488, 0, 10,  0, 52, 0, 100, 0,    2, 4488, 0, 0,  1,    4, 3000, 0,0,0,0, 7, 0,0,0,0,0,0,0,"Parqual Fintallas - on Text 2 Over - Say Text 4"),
+(4488, 0, 11,  0, 52, 0, 100, 0,    4, 4488, 0, 0, 11, 6767,    0, 0,0,0,0, 7, 0,0,0,0,0,0,0,"Parqual Fintallas - on Text 4 Over - Cast spell 6767 'Mark of Shame'"),
```
Comment (joschiwald) [inline sql/updates/world/2016_01_25_09_world.sql]: the whole sai can be simplified, this line exists 3 times

Comment (tkrokli) [inline sql/updates/world/2016_01_25_09_world.sql]: OK. I guess I tried using SMART_ACTION_LINK at first, but when that didn't work, I forgot to consider simplification.
Fixing this in a few minutes.

File: sql/updates/world/2016_01_25_09_world.sql (hunk 1 above)
Comment (tkrokli) [inline sql/updates/world/2016_01_25_09_world.sql]: Hmm, weird. I didn't notice if the default gossip got inserted in the quest text.
BTW, which text are you referring to?

Comment (joschiwald) [inline sql/updates/world/2016_01_25_09_world.sql]: 5822 which already exists in db

Comment (tkrokli) [inline sql/updates/world/2016_01_25_09_world.sql]: OK, thanks for info. I will include that condition. I just found it odd that it seems to work without a condition, as if the quest_template_addon has determined the gossip flow for the quest text.

**Comment by tkrokli** [comment]
Requested changes are implemented. Feel free to comment if something else should be changed.

**Comment by Treeston** [comment]
Gossip should only show if player is not already effected with "Mark of Shame". Gossip should also probably not show while he's talking.

**Comment by tkrokli** [comment]
OK, thanks for pointing it out to me. Have to include a condition or follow your suggestion in #trinity:
`easy way: set a data when gossip selected, reset it after punishment spell is cast`

**Comment by Treeston** [comment]
That's an "and". Set a data while he's talking, and conditions check the data to determine if you should show gossip.

**Comment by tkrokli** [comment]
OK, thanks for the extra info. Looking up how to do that kind of procedure now. :+1:

**Comment by tkrokli** [comment]
Exploits avoided by removing `npcflags` from the NPC while he talks, then restore again after spellcast.

**Pull request merged**